packages/algorand-js/src/interpreter/opcode-list.ts

```diff
@@ -1005,9 +1005,17 @@ export class Txn extends Op {
   readonly field: string;
   readonly interpreter: Interpreter;
 
-  constructor (txField: string, interpreter: Interpreter) {
+  /**
+   * @param args words list extracted from line
+   * @param line line number in TEAL file
+   * @param interpreter interpreter object
+   */
+  constructor (args: string[], line: number, interpreter: Interpreter) {
     super();
-    this.field = txField;
+    assertLen(args.length, 1, line);
+    assertOnlyDigits(args[0]);
```

Comment:
I don't think we'll be receiving digits here. (it should be a string eg. `txn Fee`). Could you please validate?

Reply:
checked with Teal opcode documentation , it mentions we will receive numbers

![wer](https://user-images.githubusercontent.com/26249903/102609042-e039b700-4150-11eb-809a-7c938e1ecc86.png)

just checked, it doesn't work with compiler.
`txn Fee` works
`txn 0` doesn't

what is this documentation ?

Reply:
`txn Fee` is processed by `assember.go` in go-algorand. Here since we have a parser, we should be processing strings. 
Seems like they documented the opcodes based on assembler input. :laughing:

Reply:
Is this documentation for their engineers or a person who wants to write code in TEAL ?

Reply:
@ratik21  - in go-algorand, the teal is compiled to a bytecode, and the assembler operates on the bytecode, not on the teal source code.

Reply:
@amityadav0  Please document the `args` - describe what is inside. (in all new constructors as well)

Reply:
> @ratik21 - in go-algorand, the teal is compiled to a bytecode, and the assembler operates on the bytecode, not on the teal source code.

ah, okay. Thanks for note.

Reply:
yes, adding args docs